index.bs

```diff
@@ -0,0 +1,505 @@
+     directories.
+
+     The |options| argument sets options that influence the behavior of the shown file picker.
+
+     |options|.{{ChooseFileSystemEntriesOptions/type}} specifies the type of the entry the website
+     wants the user to pick. When set to {{ChooseFileSystemEntriesType/"openFile"}} (the default),
+     the user can select only existing files. When set to {{ChooseFileSystemEntriesType/"saveFile"}}
+     the dialog will additionally let the user select files that don't yet exist. Finally when set
+     to {{ChooseFileSystemEntriesType/"openDirectory"}}, the dialog will let the user select
+     directories instead of files.
+
+     If |options|.{{ChooseFileSystemEntriesOptions/multiple}} is false (or absent) the user can
+     only select a single file, and the |result| will be a single {{FileSystemHandle}}. If on the
+     other hand |options|.{{ChooseFileSystemEntriesOptions/multiple}} is true, the dialog can let
+     the user select more than one file, and |result| will be an array of {{FileSystemHandle}}
+     instances (even if the user did select a single file, if
+     {{ChooseFileSystemEntriesOptions/multiple}} is true this will be returned as a single-element
+     array).
+
+     Finally |options|.{{ChooseFileSystemEntriesOptions/accepts}} and
+     |options|.{{ChooseFileSystemEntriesOptions/excludeAcceptAllOption}} specify the types of files
+     the dialog will let the user select. Each entry in
+     |options|.{{ChooseFileSystemEntriesOptions/accepts}} describes a single type of file,
+     consisting of a {{ChooseFileSystemEntriesOptionsAccepts/description}}, zero or more
+     {{ChooseFileSystemEntriesOptionsAccepts/mimeTypes}} and zero or more
+     {{ChooseFileSystemEntriesOptionsAccepts/extensions}}. Options with no valid
+     {{ChooseFileSystemEntriesOptionsAccepts/mimeTypes}} and no
+     {{ChooseFileSystemEntriesOptionsAccepts/extensions}} are invalid and are ignored. If no
+     {{ChooseFileSystemEntriesOptionsAccepts/description}} is provided one will be generated.
+
+     If |options|.{{ChooseFileSystemEntriesOptions/excludeAcceptAllOption}} is true, or if no valid
+     entries exist in |options|.{{ChooseFileSystemEntriesOptions/accepts}}, a option matching all
+     files will be included in the file types the dialog lets the user select.
+</dl></div>
+
+<div algorithm>
+The <dfn method for=Window>chooseFileSystemEntries(|options|)</dfn> method, when invoked, must run
+these steps:
+
+1. TODO
+
+</div>
+
+# Accessing special filesystems # {#special-filesystems}
+
+## The {{FileSystemDirectoryHandle/getSystemDirectory()}} method ## {#api-getsystemdirectory}
+
+<xmp class=idl>
+enum SystemDirectoryType {
+  "sandbox"
+};
+
+dictionary GetSystemDirectoryOptions {
+  required SystemDirectoryType type;
+};
+
+partial interface FileSystemDirectoryHandle {
+  static Promise<FileSystemDirectoryHandle> getSystemDirectory(GetSystemDirectoryOptions options);

```

Comment:
Hmmm... except for `sandbox` ISTM we'd want to hang this off Window instead, so operating systems can prompt the user (for permission, fallback directory, etc). Issue to track this?

Reply:
Yeah, I don't really have strong opinions on how this is exposed. Linked to the naming bikeshedding issue, as well as mentioned it perhaps being on Window. Although I don't see how the interface it is exposed on changes what operating systems can do wrt prompting?

Reply:
Just thinking if it's on Window then it's more obvious that you can't use it from serviceworkers where you can't prompt. But we need to solve that eventually and exceptions are a thing, so yeah - not a big reason to put it on Window.